internal/providers/terraform/aws/elastic_beanstalk_environment.go

```diff
@@ -0,0 +1,97 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/resources/aws"
+	"github.com/infracost/infracost/internal/schema"
+)
+
+func getElasticBeanstalkEnvironmentRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_elastic_beanstalk_environment",
+		RFunc: newElasticBeanstalkEnvironment,
+	}
+}
+
+func newElasticBeanstalkEnvironment(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+
+	var region = d.Get("region").String()
+	var StreamLogs = false
+	var DBIncluded = false
+
+	r := &aws.ElasticBeanstalkEnvironment{
+		Address: d.Address,
+		Region:  region,
+		Name:    d.Get("name").String(),
+	}
+
+	lc := &aws.LaunchConfiguration{
+		Address: "aws_launch_configuration",
+		Region:  region,
+		Tenancy: "dedicated",
+	}
+	volume := &aws.EBSVolume{
+		Address: "aws_ebs_volume",
+		Region:  region,
+	}
+	db := &aws.DBInstance{
+		Address: "aws_db_instance",
+		Region:  region,
+	}
+	cwlg := &aws.CloudwatchLogGroup{
+		Address: "aws_cloudwatch_log_group",
+		Region:  region,
+	}
+	elb := &aws.ELB{
+		Address: "aws_elb",
+		Region:  region,
+	}
+	lb := &aws.LB{
+		Address: "aws_loadbalancer",
+		Region:  region,
+	}
+
+	for _, setting := range d.Get("setting").Array() {
+		switch setting.Get("name").String() {
+		case "InstanceTypes":
+			lc.InstanceType = setting.Get("value").String()
+		case "MinSize":
+			lc.InstanceCount = intPtr(setting.Get("value").Int())
+		case "RootVolumeSize":
+			volume.Size = intPtr(setting.Get("value").Int())
+		case "RootVolumeIOPS":
+			volume.IOPS = setting.Get("value").Int()
+		case "RootVolumeType":
+			volume.Type = setting.Get("value").String()
+		case "LoadBalancerType":
+			r.LoadBalancerType = setting.Get("value").String()
+		case "StreamLogs":
+			StreamLogs = true
+		case "DBInstanceClass":
+			DBIncluded = true
+			db.InstanceClass = setting.Get("value").String()
+		case "MultiAZDatabase":
+			db.MultiAZ = true
+		case "DBEngine":
+			db.Engine = setting.Get("value").String()
+		case "DBAllocatedStorage":
+			db.AllocatedStorageGB = floatPtr(setting.Get("value").Float())
+		}
+	}
+	r.LaunchConfiguration = lc
+	r.LaunchConfiguration.RootBlockDevice = volume
+	if DBIncluded {
+		r.DBInstance = db
+	}
+	if StreamLogs {
+		r.CloudwatchLogGroup = cwlg
+	}
+	if r.LoadBalancerType == "classic" {
+		r.ElasticLoadBalancer = elb
+	} else {
+		r.LoadBalancer = lb
+	}
```

Comment:
So, something we could do to improve here is to remove `ElasticLoadBalancer` as a property and have `LoadBalancer` of type `resources.CoreResource`

we'd need to modify the `CoreResource` type to include a `BuildResource` method: 

```go
type CoreResource interface {
	PopulateUsage(u *schema.UsageData)
	BuildResource() *schema.Resource // added
}
```

this would mean that `ElasticBeanstalkEnvironment` would look like such:

```go
type ElasticBeanstalkEnvironment struct {
        ...
        LoadBalancer resources.CoreResource
        ....
}
```

Which means you could remove `LoadBalancerType` as a property and just use it as a local var here. Setting `r.LoadBalancer` with either `elb` or `lb` depending on type.

Then `PopulateUsage` and `BuildResource` in `ElasticBeanstalkEnvironment` could just call to the single property.